docs/contributing/999-rfcs/2024-02-15-resource-pinning.md

```diff
@@ -0,0 +1,159 @@
+---
+title: Resource Pinning
+description: Protection of resources from accidental deletion/moves
+---
+
+# Resource Pinning
+
+- **Author(s):**: @MarkMcCulloh
+- **Submission Date**: 2024-02-15
+- **Stage**: Proposal
+- **Stage Date**: 2024-02-15
+
+Introduces `pin` as a mechanism to protect resources from accidental deletion/moves in the logical application tree.
+
+## User Journey
+
+Sally Stateful creates a wing file with a bucket
+
+```wing
+// main.w
+bring cloud;
+
+new cloud.Bucket() as "CoolBucket";
+```
+
+because they want to make sure the bucket doesn't get deleted by accident, they pin it
+
+```wing
+// main.w
+bring cloud;
+
+pin(new cloud.Bucket() as "CoolBucket");
+```
+
+They run `wing compile` and then `wing test`, nice looks like everything works.
+
+Running `wing compile -t tf-aws` also succeeds with the following output and automatically generates a lockfile next to the entrypoint file.
+
+```shell
+# (output)
+[+pin] /root/CoolBucket
+```
+
+```json
+// main.w.lock.json
```

Comment:
```suggestion
// main.w.pin.json
```

semantically connect to `pin`

Reply:
I was assuming this lockfile could contain information other than pins eventually. That's me trying to future-proof (oops), so it's probably a dumb thing to do.


If this file must be semantically linked, perhaps the API should just be `lock` instead. WDYT?